File: src/OB.Data/Entities/Machine.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace OB.Data.Entities
{
    public class Machine
    {
        [Key]
        public int Id { get; set; }

        [Required, StringLength(64)]
        public int MAC_Address { get; set; }

        [Required, StringLength(32)]
        public int IP_Address { get; set; }

        [Required, StringLength(16)]
        public string District { get; set; }

        [Required, StringLength(16)]
        public string Precinct { get; set; }

        [Required, StringLength(16)]
        public string Status { get; set; }


        public bool IsActive { get; set; }


        // ----------------------------------------------  audit

        [Required]
        public DateTime DateCreated { get; set; }
        public DateTime? DateUpdated { get; set; }

        public DateTime DateValidated { get; set; }

    }
}

File: _assets/output_gen/jsModels/Machine.js

//---------------------------------------------------------------------------

export default class Machine {
  constructor(options) {

    this.Id = null;
    this.MAC_Address = null;
    this.IP_Address = null;
    this.District = null;
    this.Precinct = null;
    this.Status = null;
    this.IsActive = null;
    this.DateUpdated = null;
    this.UpdatedBy = null;

    if(options) {
      this.init(options);
    }
  }

  //---------------------------------------------------------------------------
  init(options) {

    this.Id = options.Id || this.Id;
    this.MAC_Address = options.MAC_Address || this.MAC_Address;
    this.IP_Address = options.IP_Address || this.IP_Address;
    this.District = options.District || this.District;
    this.Precinct = options.Precinct || this.Precinct;
    this.Status = options.Status || this.Status;
    this.IsActive = options.IsActive || this.IsActive;
    this.DateUpdated = options.DateUpdated || this.DateUpdated;
    this.UpdatedBy = options.UpdatedBy || this.UpdatedBy;

  }  
  //---------------------------------------------------------------------------  
}